### PR TITLE
Added Interval based Trigger.

### PR DESCRIPTION
**Pull request by arunpatyal**

### CODE REVIEW

File: deployments/docker-compose/docker-compose-interval.yaml [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      - CHOKIDAR_USEPOLLING=true
+      - REACT_APP_RUM_SCRIPT_URL=https://rum.sumologic.com/sumologic-rum.js
+      - REACT_APP_COFFEE_BAR_URL=http://localhost:8082/order
+      - REACT_APP_SERVICE_NAME=the-coffee-bar-frontend
+      - REACT_APP_COLLECTION_SOURCE_URL=
+      - REACT_APP_APPLICATION_NAME=tcb-local
+      - REACT_APP_DEFAULT_ATTRIBUTES={"testKey":"testValue","testListKey":["1"]}
+      - REACT_APP_PROPAGATION_CORS_URLS=[/^http:\/\/localhost:8082\/.*/,/^http:\/\/the-coffee-bar:8082\/.*/,]
+  the-coffee-machine:
+    image: public.ecr.aws/g0d6f4n6/the-coffee-bar-arun:pythonAppsv0.2.1-ot1.7.1-0.26b1
```
Comment (mat-rumian) [inline deployments/docker-compose/docker-compose-interval.yaml:78]: Looks like you used here your private image. Please set the official one

File: deployments/docker-compose/docker-compose-interval.yaml [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+      - 8083:8083
+    command: ["opentelemetry-instrument", "python3", "src/bin/the_coffee_machine.py",
+              "--host=the-coffee-machine", "--port=8083", "--machine-svc-host=machine-svc",
+              "--machine-svc-port=9090", "--spike-cron=*/2 * * * *", "--spike-duration=60", "--cpu-spike-processes=1", "--interval-based-trigger=true", "--spike-interval-days=0", "--spike-interval-hours=1"]
+    environment:
+      - OTEL_TRACES_EXPORTER=otlp_proto_http
+      - OTEL_EXPORTER_OTLP_ENDPOINT=http://otelcol:55681/v1/traces
+      - OTEL_RESOURCE_ATTRIBUTES=application=tcb-local,service.name=the-coffee-machine
+  the-cashdesk:
+    image: public.ecr.aws/g0d6f4n6/the-coffee-bar-arun:pythonAppsv0.2.1-ot1.7.1-0.26b1
```
Comment (mat-rumian) [inline deployments/docker-compose/docker-compose-interval.yaml:91]: as above ^^

File: deployments/docker-compose/docker-compose-interval.yaml [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      - 8084:8084
+    command: ["opentelemetry-instrument", "python3", "src/bin/the_cashdesk.py",
+              "--host=the-cashdesk", "--port=8084", "--calculator-host=calculator-svc", "--calculator-port=8090",
+              "--connection-string=postgresql://account:account@postgres:5432/account"]
+    environment:
+      - OTEL_TRACES_EXPORTER=otlp_proto_http
+      - OTEL_EXPORTER_OTLP_ENDPOINT=http://otelcol:55681/v1/traces
+      - OTEL_RESOURCE_ATTRIBUTES=application=tcb-local,service.name=the-cashdesk
+#  the-coffee-lover:
+#    image: public.ecr.aws/g0d6f4n6/the-coffee-bar-arun:pythonAppsv0.2.1-ot1.7.1-0.26b1
```
Comment (mat-rumian) [inline deployments/docker-compose/docker-compose-interval.yaml:104]: as above ^^

File: deployments/docker-compose/docker-compose-interval.yaml [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+              "--host=the-coffee-bar", "--port=8082", "--coffeemachine-host=the-coffee-machine",
+              "--coffeemachine-port=8083", "--cashdesk-host=the-cashdesk", "--cashdesk-port=8084"]
+ #             ,"--sweets-url=LAMBDA_URL_HERE"]
+    environment:
+      - OTEL_TRACES_EXPORTER=otlp_proto_http
+      - OTEL_EXPORTER_OTLP_ENDPOINT=http://otelcol:55681/v1/traces
+      - OTEL_RESOURCE_ATTRIBUTES=application=tcb-local,service.name=the-coffee-bar
+      - OTEL_PROPAGATORS=xray,tracecontext,baggage
+  the-coffee-bar-frontend:
+    image: public.ecr.aws/g0d6f4n6/the-coffee-bar-arun:frontend-jsv0.1.1-ot0.24.0
```
Comment (mat-rumian) [inline deployments/docker-compose/docker-compose-interval.yaml:51]: Looks like you used here your private image. Please set the official one

File: deployments/docker-compose/docker-compose-interval.yaml (hunk 1 above)
Comment (arunpatyal) [reply deployments/docker-compose/docker-compose-interval.yaml]: Fixed

File: deployments/docker-compose/docker-compose-interval.yaml (hunk 2 above)
Comment (arunpatyal) [reply deployments/docker-compose/docker-compose-interval.yaml]: Fixed

File: deployments/docker-compose/docker-compose-interval.yaml (hunk 3 above)
Comment (arunpatyal) [reply deployments/docker-compose/docker-compose-interval.yaml]: Fixed

File: deployments/docker-compose/docker-compose-interval.yaml (hunk 4 above)
Comment (arunpatyal) [reply deployments/docker-compose/docker-compose-interval.yaml]: Fixed

File: deployments/docker-compose/docker-compose-interval.yaml [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+#    command: ["node", "src/clicker.js", "http://the-coffee-bar-frontend:3000"]
+#    depends_on:
+#      - the-coffee-bar
+#      - the-coffee-bar-frontend
+#    ports:
+#      - 9222:9222
+#    expose:
+#      - 9222
+  the-coffee-bar:
+    image: public.ecr.aws/g0d6f4n6/the-coffee-bar-arun:pythonAppsv0.2.1-ot1.7.1-0.26b1
```
[the remark this answers is not in the document]
Comment (arunpatyal) [reply deployments/docker-compose/docker-compose-interval.yaml]: Fixed

**Review by mat-rumian — APPROVED** [review]
LGTM :)

**Pull request merged**